CONTRIBUTING.md

```diff
@@ -29,73 +27,70 @@ If you are contributing for the first time - we would recommend reading
-     should comment above the code exactly why it is impossible to have 100% coverage.
+2. Re-export the hook implementation and all its custom types in `src/index.ts`.
+3. Fully test your hook. The tests should include tests for both DOM and SSR environments.
+   - Hook's tests should be placed in `__tests__` subdirectory, next to the source file - `dom.ts` for DOM
+     environment, `ssr.ts` for SSR environment.
+     For example: `src/useFirstMountState/__tests__/dom.ts` and `src/useFirstMountState/__tests__/ssr.ts`.
+   - Ideally, your hook should have 100% test coverage. If that is impossible, you should leave a comment
+     in the code describing why.
    - Each hook should have at least 'should be defined' and 'should render' tests in `SSR`
      environment.
    - All utility functions should also be tested.
 4. Write docs for your hook.
    - Docs should be placed in `__docs__` sub-folder, near the source file.  
-     4ex: `src/useFirstMountState/__docs__/story.mdx`.
-   - Docs are built with storybook, to help you during writing docs - start webserver with
-     `yarn storybook:watch`.
-   - Components representing hook functionality should be placed in `example.stories.tsx` within
-     `__docs__` folder. In case file name will be missing `.stories.tsx` part - code preview won't
-     work.  
-     4ex: `src/useFirstMountState/__docs__/example.stories.tsx`.
-   - Preferred format to write the docs is MDX.
+     For example: `src/useFirstMountState/__docs__/story.mdx`.
+   - Docs are built with Storybook. You can run `yarn storybook:watch` to preview your work.
+   - Write a short example demonstrating your hook in `example.stories.tsx` within the `__docs__` folder.
+     (If the filename misses the `.stories.tsx` part, Storybook can't find your example.)
+     For example: `src/useFirstMountState/__docs__/example.stories.tsx`.
+   - Docs are written in MDX format.
      [Read more about storybook docs](https://storybook.js.org/docs/react/writing-docs/introduction).
-5. Add docs link and hook summary to the `README.md`.
-6. After all above steps are done - run `yarn lint:fix` and ensure that everything is styled by our
+5. Add a summary of the hook and a link to the docs to `README.md`.
+6. After all the above steps are done, run `yarn lint:fix` to ensure that everything is styled by our
    standards.
-7. Command `yarn new-hook myAwesomeHook` will help you create proper file structure for new hook.
+7. Command `yarn new-hook myAwesomeHook` will help you to create a proper file structure for the new hook.
 
 ## Committing
 
 ### Commit message
 
 This repo uses [semantic-release](https://github.com/semantic-release/semantic-release) and
 [conventional commit messages](https://conventionalcommits.org) so prefix your commits with `fix:`,
-`feat:`, etc., if you want your changes to appear in
+`feat:`, etc., if you want your changes to appear in the
```

Comment:
Maybe we should remove the "if" and make this more "so that your changes appear..."